File: src/main/java/br/com/kronos/backend/aplicacao/basecurricular/Direito.java
package br.com.kronos.backend.aplicacao.basecurricular;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;
import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirTamanho;

import lombok.Builder;
import lombok.Getter;

@Getter
public class Direito {
    
    private long id;
    private String descricao;
    private Long idNivel;
    private String codigo;
    private boolean ativo;
    private Boolean bncc;
    private Long idInstituicao;

	@Builder
	public Direito(long id, String descricao, Long idNivel, String codigo, boolean ativo, Boolean bncc, Long idInstituicao) {

		super();
        this.id = id;
        this.descricao = exigirTamanho(descricao, "Descrição", 1, 400);
        this.idNivel = exigirNaoNulo(idNivel, "Nível"); 
        this.codigo = exigirTamanho(codigo, "Código", 1, 5);
        this.ativo = ativo;
        if (bncc != null && bncc) {
        	this.bncc = bncc;
        	this.idInstituicao = null; 
        } else {
        	this.bncc = false;
        	this.idInstituicao = idInstituicao;     
        }
	}
}

File: poc-kronos-frontend/src/model/basecurricular/Direito.js
class Direito {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.idNivel = dados ? dados.idNivel : null;
    this.nivel = dados ? dados.nivel : null;
    this.descricao = dados ? dados.descricao : null;
    this.codigo = dados ? dados.codigo : null;
    this.ativo = dados ? dados.ativo : null;
    this.bncc = dados ? dados.bncc : null;
    this.idsCamposExperiencias = dados ? dados.idsCamposExperiencias : null;
  }

  static validations(required, maxLength) {
    return {
      descricao: { required, maxLength: maxLength(400) },
      codigo: { required, maxLength: maxLength(5) },
      idNivel: { required },
      idsCamposExperiencias: { required }
    };
  }

  static validationMessages() {
    return {
      descricao: {
        required: "Descrição é obrigatória",
        maxLength: "Descrição só aceita no máximo 400 caracteres"
      },
      codigo: {
        required: "Código é obrigatório",
        maxLength: "Código só aceita no máximo 5 caracteres"
      },
      idNivel: {
        required: "Nível é obrigatório"
      },
      idsCamposExperiencias: {
        required: "Campos Experiências é obrigatório"
      }
    };
  }
}
export default Direito;
